Src/Common/AppInsightsActivity.cs

```diff
@@ -0,0 +1,64 @@
+﻿namespace Microsoft.ApplicationInsights.Common
+{
+    using System;
+    using System.Diagnostics;
+    using System.Threading;
+
+    //this is a temporary solution that mimics System.DiagnosticSource,Activity and Correlation HTTP protocol:
+    //https://github.com/lmolkova/correlation/blob/master/http_protocol_proposal_v1.md
+    //It does not implement 
+    // - Request-Id length limitation and overflow
+    internal class AppInsightsActivity
```

Comment:
If you need this class in Base SDK - send PR against 2.4.0-beta1 base and I can create a signed build for you once that PR is approved. Or you can keep this code here for now...

Need to think of a public surface of this class though.